x-pack/plugins/apm/common/span_links.ts

```diff
@@ -0,0 +1,22 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { AgentName } from '../typings/es_schemas/ui/fields/agent';
+import { Environment } from './environment_rt';
+
+export interface SpanLinkDetails {
+  traceId: string;
+  spanId: string;
+  agentName?: AgentName;
```

Comment:
No, I mean having all the properties of a `SpanLinkDetail` element that are available when we've actually found the event under the same key. Then you can check for the existence of that property to decide how to render in the UI, because TypeScript can narrow the type.

Reply:
so something like this

```js
export interface SpanLinkDetails {
  traceId: string;
  spanId: string;
  span?: {
    agentName: AgentName;
    serviceName: string;
    duration: number;
    environment: Environment;
    transactionId: string;
    spanName?: string;
    spanSubtype?: string;
    spanType?: string;
  };
}
```

`traceId` and `spanId` will always be available.

Reply:
https://github.com/elastic/kibana/pull/126630/commits/83a23fabfe90a8935d17f758d381ff513803dfb8